web/src/app/auth/login/EmailPasswordForm.tsx

```diff
@@ -69,9 +66,9 @@ export function EmailPasswordForm({
           if (loginResponse.ok) {
             if (isSignup && shouldVerify) {
               await requestEmailVerification(values.email);
-              router.push("/auth/waiting-on-verification");
+              window.location.href = "/auth/waiting-on-verification";
```

Comment:
can we add a comment as to why we're doing this?

Reply:
it feels "wrong" to do this, so would love to have the justification here to alleviate future confusion

Reply:
also, would it be possible to just do a router.refresh() or similar?

Reply:
Sadly router.refresh doesn't trigger server side refresh of values. Could use specific <Link attributes to trigger full refresh, but we'd need to either move our providers client side or have a global refresh we use for this redirect (which feels less maintainable than just triggering global update here).